salttesting/helpers.py

```diff
@@ -1131,10 +1136,10 @@ def kill_children(_children, terminate=False, kill=False):
                 except psutil.NoSuchProcess:
                     _children.remove(child)
 
-        kill_children(children)
+        kill_children([child for child in children if child.is_running() and not any(sys.argv[0] in cmd for cmd in child.cmdline())])
```

Comment:
Look at salt's version of kill children.

I'll point it out to you in the morning if you don't find it.